chainerx_cc/examples/mnist_py/train_mnist.py

```diff
@@ -168,20 +168,20 @@ def main():
 
 def get_mnist(path, name):
     path = pathlib.Path(path)
-    x_path = path / '{}-images-idx3-ubyte.gz'.format(name)
-    y_path = path / '{}-labels-idx1-ubyte.gz'.format(name)
+    x_path = str(path / '{}-images-idx3-ubyte.gz'.format(name))
+    y_path = str(path / '{}-labels-idx1-ubyte.gz'.format(name))
 
     with gzip.open(x_path, 'rb') as fx:
         fx.read(16)  # skip header
         # read/frombuffer is used instead of fromfile because fromfile does not
         # handle gzip file correctly
         x = np.frombuffer(fx.read(), dtype=np.uint8).reshape(-1, 784)
-        x.flags.writeable = True  # TODO(beam2d): remove this workaround
+        # x.flags.writeable = True  # TODO(beam2d): remove this workaround
```

Comment:
Could you explain why this fix is needed? (What would happen without this fix?)

Reply:
This will raise a ValueError: `ValueError: cannot set WRITEABLE flag to True of this array`.

When checking the flags of the ndarray, the array does not seems to hold the data, so the array is read-only.

```
C_CONTIGUOUS : True
F_CONTIGUOUS : False
OWNDATA : False
WRITEABLE : False
ALIGNED : True
WRITEBACKIFCOPY : False
UPDATEIFCOPY : False
```

Reply:
Thanks @Guriido for this patch. Would you mind sharing the version of Python and NumPy you used to produce the above error? I tried locally with Python 3.5.1 and NumPy 1.15.4 and could not reproduce it.

Reply:
Oh, I reproduced it with NumPy 1.16.

Reply:
Also noticed that these arrays are actually copied before being modified further down. It's okay to remove these lines. Could you do that instead of commenting out?